gorm.go

```diff
@@ -376,10 +376,12 @@ func (db *DB) DB() (*sql.DB, error) {
 	connPool := db.ConnPool
 
 	if dbConnector, ok := connPool.(GetDBConnector); ok && dbConnector != nil {
-		return dbConnector.GetDBConn()
+		if sqldb, err := dbConnector.GetDBConn(); sqldb != nil && err != nil {
```

Comment:
Is `&& err== nil` here?

Reply:
I apologize for the careless mistake.
In the case where `dbConnector.GetDBConn()` returns `nil, error`, checking `sqldb != nil` alone is not sufficient to return the error properly.
Therefore, `sqldb != nil || error != nil` is the correct condition.